### ISSUE TITLE
Add option to add handlers manually

### ISSUE THREAD

**Issue by MikhailArkhipov** [matarillo/LanguageServerProtocol#1]
Thank you for the nice lightweight implementation :-)

I'd suggest making ConnectionHandlerProvider public or otherwise add a way to invoke `AddHandlers` manually. Currently one has to derive from `ServiceConnection` and override all methods in the same class. I would be more convenient if one could add handlers implemented in different classes.

This part:

```
protected ServiceConnection(Stream input, Stream output)
...
       var provider = new ConnectionHandlerProvider();
       provider.AddHandlers(Handlers, this.GetType());
}
```

**Comment by matarillo** [comment]
Thank you for your suggestion. I would like to reconsider the design of HandlerProvider first, because it deeply depends on reflections and JsonRpcMethodAttribute so far.

**Comment by MikhailArkhipov** [comment]
There is also a problem with nuget - it depends on `Matarillo.IO` but does not install the dependency. And dependency is 'ProtocolReader' - but installs `Matarillo.IO` which is confusing.

BTW, language server example is Windows-only.

**Comment by matarillo** [comment]
Thanks. Added separate issues.

**Comment by matarillo** [comment]
I just released v2.1.0, which add `Connection.RequestHandlers` and `Connection.NotificationHandlers` public properties.
